tests/functional/conftest.py

```diff
@@ -0,0 +1,15 @@
+import json
+
+import pytest
+
+
+@pytest.fixture
+def slack_annotations():
+    with open("tests/data/slack_annotations.json", encoding="utf-8") as f:
+        return json.load(f)
+
+
+@pytest.fixture
+def search_annotations():
+    with open("tests/data/search_annotations.json", encoding="utf-8") as f:
+        return json.load(f)
```

Comment:
I think these two fixtures are only used in `cli_test.py`, can we move them into that file instead of `conftest.py`?

I prefer to keep fixtures as local as possible and minimize shared or global fixtures except when they're really necessary. The reason is that shared fixtures tend to couple different tests together, and it can become difficult to change a shared fixture without breaking lots of tests.

Can we also just have the fixtures return Python dicts directly, rather than reading JSON files? I think it's more convenient to have the data inlined into the Python, rather than have to track it down in a separate file in a data directory. This will also be faster (due to not having to read from disk) and will open up the potential to use `factory_boy` to randomly generate the annotations. If we reduce the dicts to only the fields actually read by the `slack-annotations` code, that should avoid it being way too many lines.

Reply:
Not a problem, I just went with the assumption that we want to make tests more "real" but on the fence myself here.